### DISCUSSION TITLE
make execution_settings local

### DISCUSSION THREAD

**Discussion by IngridJSJ** [Spine-project/Spine-Toolbox#1734] [Discussion]
Dear Spine team, 
as we are constaltly changing the code through git, everytime I make a gitpull the execution settings are changed. 
Is there a  possibility to make the execution settings in the tool jsons as local? 
Or how could we avoid these paths to be changed everytime? 

For now, I am git ignoring the whole folder, but then the other changes won't be reflected.

**Comment by soininen** [comment]
Which execution settings are you talking about specifically?

Toolbox has recently gained the technology to separate 'local' project settings into `<project root>/.spinetoolbox/local/` directory. This is currently used to store MySQL usernames and passwords only but we are actively looking for identifying other potential 'local' settings as well.

**Reply by IngridJSJ** [reply]
I meant this  execution_settings, for example under .spinetoolbox/specification/Tool/amiris.json

 "execution_settings": {
        "env": "",
        "kernel_spec_name": "emlabpy",
        "use_jupyter_console": true,
        "executable": "C:\\Programs\\Anaconda3\\envs\\emlabpy\\python.exe"
    } 

or 

    "execution_settings": {
        "env": "conda",
        "kernel_spec_name": "conda-env-amirisspine-py",
        "use_jupyter_console": true,
        "executable": "",
        "fail_on_stderror": false
    }
Otherwise everytime there is a git pull, the interpreter or kernel spec for each tool could be overwritten. 
Or is there a way to avoid this?

**Reply by soininen** [reply]
I see what you mean. Unfortunately, there is currently no way to avoid those settings getting overwritten by git pull. It is still something we should implement in Toolbox in the future as it is pretty clear that such functionality would be highly desirable.

**Reply by soininen** [reply]
I opened an issue about this: https://github.com/Spine-project/Spine-Toolbox/issues/1735

**Reply by IngridJSJ** [reply]
Great! thank you for that!

**Reply by soininen** [reply]
The issue has been resolved now: Tool's execution settings are stored separately in project local settings. Just make sure to exclude the `<project root>/.spinetoolbox/local/` directory in the `.gitignore` file and you should be able to set up your Python tools independent of others.
